File: src/main/java/dominio/empresas/Cuenta.java
package dominio.empresas;

	import java.time.Year;
	import java.util.Objects;

	public class Cuenta {

		private Year anio;
		private String tipoCuenta;
		private int valor;

		public Cuenta(Year anio, String tipoCuenta, int valor) {
			this.anio = anio;
			this.tipoCuenta = tipoCuenta;
			this.valor = valor;
		}

		public void actualizar(int valorActualizado) {
			this.valor = valorActualizado;
		}

		public boolean esDeTipoYAnio(String tipoCuenta, Year anio) {
			return esDeTipo(tipoCuenta) && esDeAnio(anio);
		}

		private boolean esDeTipo(String tipo) {
			return this.tipoCuenta.equalsIgnoreCase(tipo);
		}

		private boolean esDeAnio(Year anio) {
			return this.anio.equals(anio);
		}

		public Year getAnio() {
			return anio;
		}

		public String getTipoCuenta() {
			return tipoCuenta;
		}

		public int getValor() {
			return valor;
		}

		@Override
		public boolean equals(Object obj) {
			return (obj instanceof Cuenta) && Objects.equals(anio, ((Cuenta) obj).getAnio())
					&& Objects.equals(tipoCuenta, ((Cuenta) obj).getTipoCuenta());
		}

		@Override
		public int hashCode() {
			return Objects.hash(anio, tipoCuenta);
		}

	}

File: src/main/java/dominio/empresas/Cuenta.kt
package dominio.empresas

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.Table
import java.time.Year
import java.util.Objects

@Entity
@Table(name = "cuentas")
class Cuenta(val anio: Year, val tipoCuenta: String, valor: Int) {

    @Id
    @GeneratedValue
    private val id: Long? = null

    var valor: Int = valor
        private set

    fun actualizar(cuentaConDatosNuevos: Cuenta) {
        valor = cuentaConDatosNuevos.valor
    }

    fun esDeTipo(tipo: String): Boolean {
        return this.tipoCuenta.equals(tipo, ignoreCase = true)
    }

    fun esDeAnio(anio: Year): Boolean {
        return this.anio == anio
    }

    override fun equals(other: Any?): Boolean {
        return other is Cuenta &&
                anio == other.anio &&
                tipoCuenta == other.tipoCuenta
    }

    override fun hashCode(): Int {
        return Objects.hash(anio, tipoCuenta)
    }

}